File: Leetcode/TwoPointers/reversePrefix.js
/**
 * @param {string} w
 * @param {character} c
 * @return {string}
 */
function reversePrefix(w, c) {
    let sp = 0;
    let p = 0;
    let s = '';
    while (sp < w.length) {
        if (w[sp] !== c) {
            p++
            sp++
        } else {
            break;
        }
    }

    if (sp == w.length) {
        console.log("Target char doesn't exist");
        return w;
    }

    while (p >= 0) {
        s+= w[p];
        p--;
    }

    sp++;
    while (sp < w.length) {
        s+= w[sp]
        sp++;
    }
    console.log("Result: ", s);
    return s;
};

reversePrefix("abcd", "z");

File: Leetcode/TwoPointers/ReversePrefix.java
package Leetcode.TwoPointers;

public class ReversePrefix {
    static public void main(String[] args) {
        System.out.println("Reversing string:");
        String word = "abcd";
        char target = 'z';
        System.out.println(word);
        System.out.println("Target:");
        System.out.println(target);
        String ans = reversePrefixStr(word, target);
        System.out.println("Result:");
        System.out.println(ans);
    }

    static private String reversePrefixStr(String w, char c) {
        int sp = 0;
        int p = 0;

        while (sp < w.length() - 1) {
            if (w.charAt(sp) != c) {
                p++;
                sp++;
            } else {
                break;
            }
        }
        if (sp == w.length() - 1) {
            System.out.println("Target character not found");
            return w;
        }

        StringBuilder finalString = new StringBuilder();
        while (p >= 0) {
            finalString.append(w.charAt(p));
            p--;
        }

        sp++;
        while (sp < w.length()) {
            finalString.append(w.charAt(sp));
            sp++;
        }

        return finalString.toString();
    }
}
